pkg/kvstore/etcd.go

```diff
@@ -907,7 +912,9 @@ reList:
 					// recreate the watcher and try to
 					// watch on the next possible revision
 					if errors.Is(err, v3rpcErrors.ErrCompacted) {
-						scopedLog.WithError(Hint(err)).Debug("Tried watching on compacted revision")
+						scopedLog.WithError(Hint(err)).Info("Tried watching on compacted revision. Triggering relist of all keys")
+					} else {
+						scopedLog.WithError(Hint(err)).Warning("Etcd watcher errored. Triggering relist of all keys")
```

Comment:
This turned out not being a great idea, as it is also triggered by the clustermesh interceptor when the target clustermesh-apiserver is restarted, in turn getting flagged by CI. Another observed errors is:

```
level=warning msg="Etcd watcher errored. Triggering relist of all keys" config=/var/lib/cilium/clustermesh/cluster1 endpoints="[https://clustermesh-apiserver.kube-system.svc:2379/]" error="rpc error: code = Unknown desc = malformed header: missing HTTP content-type" prefix=cilium/cache/identities/v1/cluster1/id revision=0 subsys=kvstore (1 occurrences)
``` 

I'll try to figure out whether it is possible to ignore these specific cases, or the warning should be demoted/dropped altogether.

Reply:
Updated that piece of code to only emit a debug log if the error is expected and triggered by an interceptor (I used an extra error type to avoid leaking details of the interceptor into the kvstore implementation). Additionally lowered the severity of generic errors to info level, given that they could potentially happen in clustermesh scenarios when the clustermesh-apiserver is restarted, to avoid issues in CI.

Reply:
> Additionally lowered the severity of generic errors to info level, given that they could potentially happen in clustermesh scenarios when the clustermesh-apiserver is restarted, to avoid issues in CI.

I'm not sure about it. If remote service (clustermesh-apiserver) restarts I would definitely expect to have a warning log instead of info log. This is not part of "normal" operations, but we expect it to resolve by itself.

Reply:
Yeah, I don't terribly like it either. Having it at info level just seemed a decent compromise to me to avoid further increasing the complexity of this PR with trying to figure out all expected cases to demote, considering that it is already an improvement compared to the existing version which doesn't emit any logs in that case.

> If remote service (clustermesh-apiserver) restarts I would definitely expect to have a warning log instead of info log. This is not part of "normal" operations, but we expect it to resolve by itself.

There is actually already a warning log that covers that case (allow-listed in CI), and that is emitted by the clustermesh subsystem when the etcd connection is restarted: https://github.com/cilium/cilium/blob/a3489f190ba6e87b5336ee685fb6c80b1270d06d/pkg/clustermesh/common/remote_cluster.go#L251